### DISCUSSION TITLE
PLEASE HELP firewall commands not working

### DISCUSSION THREAD

**Discussion by Phillip-Samson** [Security-Onion-Solutions/securityonion#11041] [2.4] [answered]
2.4 RC2
Manager/Search/Forward Node
Oracle 9.2
I have installed multiple integrations into the so-grid-nodes_general policy that my Forward Node is a member of and whenever I am sending logs from an appliance such as Palo or Cisco or Squid proxy i see logs coming into the forward node via tcpdump but they seem to never end up in an index. 
As I read the instructions my understanding is that the integration is supposed to configure the Elastic_agent to receive the logs. 
Am I wrong do I need to go into the advanced setting of the firewall and define these new sources when i attempt to add new host and port groups they fail  when I try and run the so-firewall add port group commands that feature is not available 
if I add the port number to syslog or senor in the administrator config firewall setting in the console  and point the device to the manager search node it still will not work
please point me to the proper documentation 
I am completely confused about how to get these integrations working.

I am unable to add hostgroups or portsgroups
<img width="1059" alt="image" src="https://github.com/Security-Onion-Solutions/securityonion/assets/36277967/0a514b93-8f76-4856-b292-4cde4153ab9c">

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/36277967/18bbbee3-482f-4856-990e-859ba16d49ca)

Allow hosts to send syslog to a sensor node
By default, if you go to [Administration](https://sologging.cliffsnet.com/docs/administration.html#administration) –> Configuration –> firewall and add a host to the syslog hostgroup, that host will only be allowed to connect to the manager node. If we want to allow a host or group of hosts to send syslog to a sensor, then we can do the following:

Create a new host group that will contain the IPs of the hosts that you want to allow to connect to the sensor. This will add the host group to /opt/so/saltstack/local/salt/firewall/hostgroups.local.yaml. If the host group already exists, you can skip to step 2. Run the following on the manager:
sudo so-firewall addhostgroup <GROUP_NAME>
Add the desired IPs to the host group. This will add the IPs to the host group in /opt/so/saltstack/local/salt/firewall/hostgroups.local.yaml.
sudo so-firewall includehost <GROUP_NAME> <IP>

**Comment by TOoSmOotH** [comment]
The docs are not correct. We are in the process of updating them. Remove the file you put in local. There are pre-configured custom hostgroups and portgroups in the UI that you will need to use. If you choose the advanced option in the config UI you will see the groups named customhostrgroupX and customportgroupX.

**Reply by Phillip-Samson** [reply]
Do I rename the customgroups?

**Reply by TOoSmOotH** [reply, marked as answer]
Here is the proposed changes to the document:

Creating a custom host group with a custom port group
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

In this example, we will be adding a new custom hostgroup to allow a custom set of hosts to connect to a custom port on the idh nodes on port 1234.

1. Select "Show all configuration settings, including advanced settings." on the options dropdown menu.
2. Under firewall select customhostgroup0 that is part of the hostgroups section.
3. Select the idh node that you want to allow access to under the dropdown.
4. Add the list of hosts that require access and select the checkmark.
5. Under firewall select customportgroup0 that is part of the portgroups section.
6. Select the idh node that you want to allow access to under the dropdown.
7. Add the apporpriate port under the appropriate protocol. In this case we will be adding 1234 tcp and selecting the checkmark.
8. Under firewall/roles select idh/chain/DOCKER-USER/hostgroups/customhostgroups/portgroups.
9. Select the idh node that we want to allow access to under the dropdown. 
10. Add the portgroup customportgroup0 to the list and select the checkmark.
11. The next time the idh node checks in it will get the appropriate firewall rules.

**Reply by dougburks** [reply]
The firewall page in the documentation has been updated:
https://docs.securityonion.net/en/2.4/firewall.html

**Reply by Phillip-Samson** [reply]
Thank you for your quick response!
